fineract-provider/src/main/java/org/apache/fineract/portfolio/account/jobs/executestandinginstructions/ExecuteStandingInstructionsTasklet.java

```diff
@@ -97,7 +97,8 @@ public RepeatStatus execute(StepContribution contribution, ChunkContext chunkCon
                     transactionAmount = standingInstructionDuesData.totalDueAmount();
                 }
                 if (recurrenceType.isDuesRecurrence()) {
-                    isDueForTransfer = LocalDate.now(DateUtils.getDateTimeZoneOfTenant()).equals(standingInstructionDuesData.dueDate());
+                    isDueForTransfer = recurrenceType.isDuesRecurrence() && standingInstructionDuesData.dueDate() != null
```

Comment:
i am not sure the job should pick up "earlier" standing instructions than the today's date.
@bharathc27 Could you please advise here?

Reply:
Please add an integration test which covers this changed logic.